src/runtime/relax_vm/cuda/cuda_graph_builtin.cc

```diff
@@ -82,6 +96,43 @@ struct CUDAGraphCapturedState {
   cudaGraphExec_t exec = nullptr;
 };
 
+class ScopedCUDAStream {
+ public:
+  ScopedCUDAStream() { CUDA_CALL(cudaStreamCreate(&stream_)); }
+  ~ScopedCUDAStream() { cudaStreamDestroy(stream_); }
+  ScopedCUDAStream(const ScopedCUDAStream&) = delete;
+  ScopedCUDAStream(ScopedCUDAStream&&) = delete;
+  ScopedCUDAStream& operator=(const ScopedCUDAStream&) = delete;
+  ScopedCUDAStream& operator=(ScopedCUDAStream&&) = delete;
+
+  operator cudaStream_t() const { return stream_; }
+
+ private:
+  cudaStream_t stream_;
+};
+
+class CUDACaptureStream {
+ public:
+  explicit CUDACaptureStream(cudaGraph_t* graph)
+      : prev_default_stream_(CUDAThreadEntry::ThreadLocal()->stream), output_graph_(graph) {
+    CUDAThreadEntry::ThreadLocal()->stream = capture_stream_;
+
+    CUDA_CALL(cudaStreamBeginCapture(capture_stream_, cudaStreamCaptureModeGlobal));
+  }
+  ~CUDACaptureStream() {
+    cudaStreamEndCapture(capture_stream_, output_graph_);
```

Comment:
this should be enclosed with `CUDA_CALL` to check return code

Reply:
I tested it with `CUDA_CALL` to verify, and using the `CUDA_CALL` would re-introduce the same bug that this change is intended to fix.  If the stack is unwinding due to a thrown exception, then throwing another exception would result in `std::terminate` being called.  To avoid this, destructors shouldn't throw exceptions ([stackoverflow link](https://stackoverflow.com/a/130123)).

There are some ways to use `std::uncaught_exceptions` to determine whether an exception is being unwound, and to conditionally throw an exception if it isn't already the case.  However, those tend to be pretty context-dependent, and probably aren't worth using in this case.